kuwala/core/canvas/src/components/Modals/BlockCatalog/ExportCatalog/ExportCatalogModal.js

```diff
@@ -0,0 +1,95 @@
+import React, {Fragment, useEffect, useState} from "react";
+import {useStoreActions} from "easy-peasy";
+import ExportCatalogDTO from "../../../../data/dto/ExportCatalogDTO";
+import {getAllExportCatalogCategories, getAllItemsInExportCategory} from "../../../../api/ExportCatalog";
+import ExportCatalogFooter from "./ExportCatalogFooter";
+import ExportCatalogSelector from "./ExportCatalogSelector";
+import ExportCatalogBody from "./ExportCatalogBody";
+
+export default () => {
+    const { toggleBlockCatalogModal } = useStoreActions(actions => actions.common);
+    const [selectedExportIndex, setSelectedExportIndex] = useState(null);
+    const [catalogCategories, setCatalogCategories] = useState([]);
+    const [catalogOptions, setCatalogOptions] = useState([]);
+    const [selectedCatalogOption, setSelectedCatalogOption] = useState(null);
+
+    useEffect(() => {
+        initExportCatalogs().then(null);
+    }, [])
+
+    useEffect(()=> {
+        if(catalogCategories.length) {
+            const category = catalogCategories[selectedExportIndex]
+            fetchCatalogBodyItems(category).then(null);
+        }
+    }, [selectedExportIndex])
```

Comment:
You should to add `catalogCategories` as a dependency if you use it inside a `useEffect`.

Also all function you declare in it should be either declared inside `useEffect` and all used variables added to the dependencies, or you wrap `fetchCatalogBodyItems` with `useCallback`.

I did this refactoring to fix a bug in the `DataBlockConfigModal` on my current working branch:

``` js
    const fetchSchema = useCallback(async () => {
        if (selectedElement) {
            setIsSchemaLoading(true);

            const res = await getSchema(selectedElement.data.dataSource.id);

            if (res.status === 200) {
                const populatedSchema = populateSchema(res.data, selectedElement.data.dataSource);

                setSchema(populatedSchema)
            }

            setIsSchemaLoading(false)
        }
    }, [selectedElement])
```

Your IDE should give you warnings for this.

Reply:
refactored, moved function declaration inside useEffect.

Reply:
Still looks the same to me

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/28758040/171024731-92b8753d-dea3-4adb-ad2e-4c883ae931f9.png">